docs/best-practice/worker-head-reconnection.md

```diff
@@ -0,0 +1,37 @@
+# Explaination and Best Practice for workers-head Reconnection
+
+## Problem
+
+For a `RayCluster` with a head and several workers, if a worker is crashed, it will be relaunched immediately and re-join the same cluster quickly; however, when the head is crashed, it will run into the issue [#104](https://github.com/ray-project/kuberay/issues/104) that all worker nodes are lost from the head for a long period of time. 
+
+
+## Explaination
+
+When the head pod was deleted, it will be recreated with a new IP by KubeRay controller，and the GCS server address is changed accordingly. The Raylets of all workers will try to get GCS address from Redis in ‘ReconnectGcsServer’, but the redis_clients always use the previous head IP, so they will always fail to get new GCS address. The Raylets will not exit until max retries are reached. There are two configurations determining this long delay:
+
+```
+/// The interval at which the gcs rpc client will check if gcs rpc server is ready.
+RAY_CONFIG(int64_t, ping_gcs_rpc_server_interval_milliseconds, 1000)
+
+/// Maximum number of times to retry ping gcs rpc server when gcs server restarts.
+RAY_CONFIG(int32_t, ping_gcs_rpc_server_max_retries, 600)
+
+https://github.com/ray-project/ray/blob/master/src/ray/gcs/gcs_client/gcs_client.cc#L294-L295
```

Comment:
since master branch already changes, let's use the permalink instead?
![image](https://user-images.githubusercontent.com/4739316/152307143-ee789b6b-47a8-4734-9980-818e06b1b7e8.png)

Reply:
Changed to permalink.